clientconn.go

```diff
@@ -1234,6 +1230,7 @@ func (cc *ClientConn) ResetConnectBackoff() {
 
 // Close tears down the ClientConn and all underlying connections.
 func (cc *ClientConn) Close() error {
+	defer func() { <-cc.csMgr.pubSub.Done() }()
```

Comment:
Optional: Maybe a single `defer` instead of two? Even when we know that deferred funcs are executed LIFO, this is easier to read than having two defers I feel.

```golang
defer func() {
  cc.cancel()
  <-cc.csMgr.pubSub.Done()
}()
```

Reply:
SGTM - done